controllers/config.go

```diff
@@ -0,0 +1,231 @@
+	   Kind:       "StorageCluster",
+	   PkgNames:   []string{OcsSubscriptionPackage},
+	   -----------------------------------------------
+	   CrdName:    "cephclusters.ceph.rook.io",
+	   ApiVersion: "ceph.rook.io/v1",
+	   Kind:       "CephCluster",
+	   PkgNames:   []string{RookSubscriptionPackage, CephCSISubscriptionPackage, CSIAddonsSubscriptionPackage},
+	*/
+
+	CrdName    string
+	ApiVersion string
+	Kind       string
+	CrdPresent bool
+	PkgNames   []string
+}
+
+func GetCrdAndPackagesMappingList(ctx context.Context, cli client.Client, logger logr.Logger, operatorNamespace string) ([]CrdAndPackages, error) {
+
+	configMapData, err := GetConfig(ctx, cli, logger, operatorNamespace)
+	if err != nil {
+		logger.Error(err, "failed to get configmap data")
+		return nil, err
+	}
+
+	// build a mapping of CRD names to the corresponding package names
+	// this allows us to associate each CRD with the multiple packages
+	crdToPkgNames := map[string][]string{}
+	for _, value := range configMapData {
+		for _, crd := range value.ScalerCrds {
+			crdToPkgNames[crd] = append(crdToPkgNames[crd], value.Pkg)
+		}
+	}
+
+	// create a slice of ResourceMappingRecord
+	resourceMapping := []CrdAndPackages{}
+	for crd, pkgNames := range crdToPkgNames {
+		resourceMapping = append(resourceMapping, CrdAndPackages{
+			CrdName:  crd,
+			PkgNames: pkgNames,
+		})
+	}
+
+	// get the CRD details for each CRD (API version and Kind)
+	for i, record := range resourceMapping {
+		crd := &extv1.CustomResourceDefinition{}
+		if err := cli.Get(ctx, client.ObjectKey{Name: record.CrdName}, crd); err != nil {
+			if errors.IsNotFound(err) {
+				logger.Error(err, "CRD not found", "crdName", record.CrdName)
+				continue
+			}
+			logger.Error(err, "failed to get CRD", "crdName", record.CrdName)
+			return nil, err
+		}
+		resourceMapping[i].CrdPresent = true
+		resourceMapping[i].ApiVersion = crd.Spec.Group + "/" + crd.Spec.Versions[0].Name
+		resourceMapping[i].Kind = crd.Spec.Names.Kind
+	}
+
+	logger.Info("parsed resource mapping successfully", "resourceMapping", resourceMapping)
+
+	return resourceMapping, nil
+}
+
+func GetCrdNamesMap(ctx context.Context, cli client.Client, logger logr.Logger, operatorNamespace string) (map[string]bool, error) {
+
+	configMapData, err := GetConfig(ctx, cli, logger, operatorNamespace)
+	if err != nil {
+		logger.Error(err, "failed to get configmap data")
+		return nil, err
+	}
+
+	crdNames := map[string]bool{}
+	for _, value := range configMapData {
+		for _, crd := range value.ScalerCrds {
+			crdNames[crd] = true
+		}
+	}
+
+	logger.Info("parsed CRD names successfully", "crdNames", crdNames)
+
+	return crdNames, nil
+}
+
+func GetCsvNamesMap(ctx context.Context, cli client.Client, logger logr.Logger, operatorNamespace string) (map[string]bool, error) {
+
+	configMapData, err := GetConfig(ctx, cli, logger, operatorNamespace)
+	if err != nil {
+		logger.Error(err, "failed to get configmap data")
+		return nil, err
+	}
```

Comment:
This need to be provided from the outside as there is no point in reading and parsing the same source map in multiple places